java/ruleXtract/src/main/java/uk/ac/cam/eng/extraction/hadoop/util/SimpleHFileOutputFormat.java

```diff
@@ -42,27 +44,27 @@
  * @date 28 May 2014
  */
 public class SimpleHFileOutputFormat extends
-		FileOutputFormat<Text, TargetFeatureList> {
+		FileOutputFormat<RuleString, TargetFeatureList> {
 
 	@Override
-	public RecordWriter<Text, TargetFeatureList> getRecordWriter(
+	public RecordWriter<RuleString, TargetFeatureList> getRecordWriter(
 			TaskAttemptContext job) throws IOException {
 
 		final Configuration conf = job.getConfiguration();
 		Path file = getDefaultWorkFile(job, ".hfile");
 		FileSystem fs = file.getFileSystem(conf);
-		HFile.WriterFactory writerFactory = HFile.getWriterFactory(conf);
-		final HFile.Writer writer = writerFactory.createWriter(fs, file,
-				64 * 1024, "gz", null);
 		final CacheConfig cacheConfig = new CacheConfig(conf);
-		return new RecordWriter<Text, TargetFeatureList>() {
+		HFile.WriterFactory writerFactory = HFile.getWriterFactory(conf, cacheConfig).withPath(fs, file)
+				.withBlockSize(64 * 1024).withCompression("gz");
+		final HFile.Writer writer = writerFactory.create();
+		return new RecordWriter<RuleString, TargetFeatureList>() {
 
 			private ByteArrayOutputStream bytesOut = new ByteArrayOutputStream();
 
 			private DataOutputStream out = new DataOutputStream(bytesOut);
 
 			BloomFilterWriter bloomFilterWriter = BloomFilterFactory
-					.createBloomAtWrite(conf, cacheConfig, BloomType.ROW, -1,
+					.createGeneralBloomAtWrite(conf, cacheConfig, BloomType.ROW, -1,
```

Comment:
I think it's reasonably obvious that it is not meant to be a meaningful value. This is an HBase method call and they have not provided a constant. I don't want to start documenting the HBase api. If you use eclipse then you can get a description of the method arguments by hovering over the method with your cursor.